src/Dapr.Jobs/DaprJobsGrpcClient.cs

```diff
@@ -0,0 +1,312 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Text.Json;
+using Dapr.Jobs.Models.Responses;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using Grpc.Net.Client;
+using Autogenerated = Dapr.Client.Autogen.Grpc.v1;
+
+namespace Dapr.Jobs;
+
+/// <summary>
+/// A client for interacting with the Dapr endpoints.
+/// </summary>
+internal sealed class DaprJobsGrpcClient : DaprJobsClient
+{
+    private readonly Uri httpEndpoint;
+    internal readonly HttpClient httpClient;
+    
+    private readonly GrpcChannel channel;
+    private readonly Autogenerated.Dapr.DaprClient client;
+    internal readonly KeyValuePair<string, string>? apiTokenHeader;
+
+    // property exposed for testing purposes
+    internal Autogenerated.Dapr.DaprClient Client => client;
+    
+    internal DaprJobsGrpcClient(
+        GrpcChannel channel,
+        Autogenerated.Dapr.DaprClient innerClient,
+        HttpClient httpClient,
+        Uri httpEndpoint,
+        JsonSerializerOptions _,
```

Comment:
Nit: just omit the `JsonSerializerOptions` argument?

Reply:
Done. Also removed `httpEndpoint` as it's not used anywhere either (since this just uses gRPC).